core/common/src/main/scala/org/typelevel/otel4s/meta/InstrumentMeta.scala

```diff
@@ -44,4 +47,9 @@ object InstrumentMeta {
       val unit: F[Unit] = Applicative[F].unit
     }
 
+  private class MappedK[F[_], G[_]](meta: InstrumentMeta[F])(f: F ~> G)
```

Comment:
I like this more than `implicit class XXXSyntax[F[_]] { def mapK[G[_]](fk: F ~> G): ... } `